-----------------------------------------------------------------
creating errors with err:= error.New()
-------------------------------------------------------------------
package main

import (
	"errors"
	"fmt"
)

func main() {
	err := errors.New("barnacles")
	fmt.Println("Sammy says:", err)
}

-----------------------------------------------------------------------------
with   err:= fmt.Errorf()
----------------------------------------------------------------------------------
package main

import ("fmt"
	"time")

func main() {
	err := fmt.Errorf("error occurred at: %v", time.Now())
	fmt.Println("An error happened:", err)
}

-------------------------------------------------------------------------------------------------------------------
Handling error
-------------------------------------------------------------------------------------
package main

import ("errors"
	"fmt")

func boom() error {
	return errors.New("barnacles")
}

func main() {
	err := boom()

	if err != nil {
		fmt.Println("An error occurred:", err)
		return
	}
	fmt.Println("Anchors away!")
}

